src/app/clusters/test-cluster-server/test-cluster-server.cpp

```diff
@@ -165,30 +222,49 @@ EmberAfStatus writeTestListOctetAttribute(EndpointId endpoint)
 CHIP_ERROR TestAttrAccess::ReadListOctetStringAttribute(AttributeValueEncoder & aEncoder)
 {
     return aEncoder.EncodeList([](const TagBoundEncoder & encoder) -> CHIP_ERROR {
-        constexpr uint16_t attributeCount = 4;
-        char data[6]                      = { 'T', 'e', 's', 't', 'N', '\0' };
-
-        for (uint8_t index = 0; index < attributeCount; index++)
+        for (uint8_t index = 0; index < kAttributeCount; index++)
         {
-            snprintf(data + strlen(data) - 1, 2, "%d", index);
-            ByteSpan span(Uint8::from_char(data), strlen(data));
+            ByteSpan span(Uint8::from_char(listOctetStringData[index]), strlen(listOctetStringData[index]));
             ReturnErrorOnFailure(encoder.Encode(span));
         }
         return CHIP_NO_ERROR;
     });
 }
 
+CHIP_ERROR TestAttrAccess::WriteListOctetStringAttribute(TLV::TLVReader & aReader)
+{
+    DataModel::DecodableList<ByteSpan> list;
+
+    ReturnErrorOnFailure(DataModel::Decode(aReader, list));
+
+    uint8_t index = 0;
+    auto iter     = list.begin();
+    while (iter.Next())
+    {
+        const ByteSpan & entry = iter.GetValue();
+
+        Platform::CopyString(listOctetStringData[index], sizeof(listOctetStringData[index]), entry);
```

Comment:
Buffer-overflow. Please add error checking if index exceeds the size of this list.